subsys/bluetooth/host/scan.c

```diff
@@ -716,22 +717,30 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 			      BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
 	     ((pending_per_adv_sync->sid != evt->sid) ||
 	      bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
-		struct bt_le_per_adv_sync_term_info term_info;
-
 		BT_ERR("Unexpected per adv sync established event");
+		/* Request terminate of pending periodic advertising in controller */
 		per_adv_sync_terminate(sys_le16_to_cpu(evt->handle));
 
+		unexpected_evt = true;
```

Comment:
In the case that the event is unexpected, the evt->status would still be BT_HCI_ERR_SUCCESS, which doesn't supply anything to the upper layer that gets the callback. 

Should we add an error code, even just `BT_HCI_ERR_UNSPECIFIED`, to the callback in this case? 

The same comment somewhat applies for `bt_hci_le_per_adv_sync_lost` as well, where the sync lost reason/error is not supplied by the controller, but where a `0`/`BT_HCI_ERR_SUCCESS` does not really provide anything for the upper layer.

Reply:
Yeah, I searched for something like that but I didn't find any useful error name. I also thought that these errors should be taken from BT Core spec.

Reply:
The `BT_HCI_ERR_*` values are from the core spec, and as such are those that should be used. 

Perhaps `BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER` is fitting?

Reply:
Ah... ok then I would suggest to go with `BT_HCI_ERR_UNSPECIFIED` instead of re-use of other. It looks like the situation is not mentioned in the BT Code so there is no assigned error code.

From BT Core: 
> The Unspecified Error error code indicates that no other error code specified is
appropriate to use.

Reply:
Or add dedicated error codes for terminate callback. This is actually up to host implementation and it does not depend on BT Core. Host correctly handles cases from BT Core. How it handles it later, I mean between itself and an application is dependent on implementation.

Reply:
I've added BT_HCI_ERR_UNSPECIFIED but I'm open to use dedicated error codes.